library/swift/RequestMethod.swift

```diff
@@ -0,0 +1,36 @@
+import Foundation
+
+/// Represents an HTTP request method.
+@objc
+public enum RequestMethod: Int {
```

Comment:
If we implement `RawRepresentable` we can use this more ergonomically in Swift, I think.(?)

Reply:
I mentioned this in the PR description, but unfortunately we have to use a variant of `Int` instead of `String` as the raw value in order for this to be visible to Objective-C

Reply:
Discussed further offline, and something like what I've pasted below would work. However I don't really think this gives us much benefit:
- We end up with more code because we have to do 2-way mappings from `String`
- We probably don't want to expose the raw `String` values publicly (which would be required here) - they are only needed for communicating to the C library

Thoughts?

```swift
@objc
enum Foo: Int, RawRepresentable {
    case abc

    typealias RawValue = String

    init?(rawValue: String) {
        switch rawValue {
        case "abc":
            self = .abc
        default:
            return nil
        }
    }

    var rawValue: String {
        switch self {
        case .abc:
            return "abc"
        }
    }
}
```

Reply:
To clarify on my emoji, I agree with your take that this doesn't actually end up benefiting us a lot. :) I was thinking it would let us get implicit casts to Strings with the enum values, and if it doesn't, there's no real reason to go down this path.

Reply:
Sounds good!